fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java

```diff
@@ -58,7 +60,11 @@ public static void analyze(UpdateStmt updateStmt, ConnectContext session) {
                 if (col.isKey()) {
                     throw new SemanticException("primary key column cannot be updated: " + col.getName());
                 }
-                item = new SelectListItem(new CastExpr(col.getType(), assign.getExpr()), col.getName());
+                if (assign.getExpr() instanceof LiteralExpr) {
+                    item = new SelectListItem(TypeManager.addCastExpr(assign.getExpr(), col.getType()), col.getName());
```

Comment:
Could you add comment why call TypeManger.addCastExpr for LiteralExpr?
And why not call `TypeManager.addCastExpr` for other expressions?